x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/ExpressionBuilder.java

```diff
@@ -222,17 +235,18 @@ public LikePattern visitPattern(PatternContext ctx) {
         }
 
         char escape = 0;
-        String escapeString = string(ctx.escape);
+        PatternEscapeContext escapeCtx = ctx.patternEscape();
+        String escapeString = string(escapeCtx != null ? escapeCtx.escape : null);
```

Comment:
`escapeCtx == null ? null : string(escapeCtx.escape);` feels a little cleaner to me. No need to pitch `null` through `string` and make the reader wonder what `string` does with null.

Reply:
+1